### ISSUE TITLE
[Bug]: PadTo is probably misnamed, since it crops if the image is too large.

### ISSUE THREAD

**Issue by bw4sz** [kornia/kornia#3518]
### 🐛 Describe the bug

Issue: PadTo crops images when they are larger than the target size,
instead of only padding when smaller. 

Expected behavior: PadTo should only pad (add borders), never crop.
When image is larger than target, it should remain unchanged.

This might be just a case where the name or docs should be fixed. I think pad always suggest getting bigger. The behavior here is like resize + pad. This matters for object detection pipeline, since you might silently lose alot of data outside the bounds for images of mixed sizes.

### 🔄 Steps to Reproduce

```bash
1. Load the module
2. Try PadTo on an image that is larger than the target size
3. Confirm that the image has been cropped to that size.
```

### 💻 Minimal Code Example

```python
import torch
import kornia.augmentation as K
from kornia.constants import DataKey

# Create a test image larger than target size
# Image: 1100 x 1100 (larger than target 1000 x 1000)
image = torch.randn(1, 3, 1100, 1100)  # (B, C, H, W)

# Create bounding boxes in original image coordinates
# Box at bottom-right corner: (700, 1000, 800, 1100)
# This box will be out of bounds after cropping to 1000x1000
boxes = torch.tensor([[[700.0, 1000.0, 800.0, 1100.0]]])  # (B, N, 4) in xyxy format

print("=" * 60)
print("Kornia PadTo Issue - Minimal Reproducible Example")
print("=" * 60)
print(f"\nInput image shape: {image.shape}")
print(f"Target size: (1000, 1000)")
print(f"Input boxes: {boxes[0, 0].tolist()}")
print(f"  Box coordinates: xmin={boxes[0,0,0]:.0f}, ymin={boxes[0,0,1]:.0f}, "
      f"xmax={boxes[0,0,2]:.0f}, ymax={boxes[0,0,3]:.0f}")

# Apply PadTo augmentation
pad_transform = K.PadTo(size=(1000, 1000))
aug_seq = K.AugmentationSequential(
    pad_transform, data_keys=[DataKey.IMAGE, DataKey.BBOX_XYXY]
)

# Apply transformation
augmented_image, augmented_boxes = aug_seq(image, boxes)

print(f"\nOutput image shape: {augmented_image.shape}")
print(f"Output boxes: {augmented_boxes[0, 0].tolist()}")
print(f"  Box coordinates: xmin={augmented_boxes[0,0,0]:.0f}, "
      f"ymin={augmented_boxes[0,0,1]:.0f}, "
      f"xmax={augmented_boxes[0,0,2]:.0f}, "
      f"ymax={augmented_boxes[0,0,3]:.0f}")
```

### ✅ Expected behavior

I would expect no padding if image_size > target_size. So the output image should remain 1100 pixels. I welcome debate if there is some functionality here I missed. 

### ❌ Actual behavior

Crops to target size, the image size is 1000, since this is target size. 

### 🔧 Environment

```shell
>>> kornia.__version__
'0.8.2'
```

### 📝 Additional context

Thank you for your supports and open source efforts.

### 🤝 Contribution Intent

- [ ] I plan to submit a PR to fix this bug
- [x] I'm reporting this bug but not planning to fix it

**Comment by Aryanpagaria** [comment]
@bw4sz  1
would be interested in working on this, if the maintainers agree that the current behavior is unintended or at least confusing given the name PadTo.


From my understanding, there are two reasonable interpretations, and I would like guidance on which direction you prefer:


Option 1: Change behavior (preferred for detection pipelines)


Modify PadTo so that it only pads when the input spatial size is smaller than the target size.

If the image is already larger than or equal to the target size in a given dimension, it remains unchanged along that dimension.

This would make PadTo a pure padding operator, aligning with its name and avoiding silent data loss (especially critical for object detection and segmentation pipelines).

Option 2: Keep behavior but clarify intent
Keep the current implementation (crop-or-pad to fixed size).

Rename the transform (e.g., PadOrCropTo) or

Update the documentation very explicitly to state that PadTo enforces a fixed output size and may crop larger inputs.

Implementation approach (if Option 1 is accepted)
In the forward pass, compute padding values as max(target - input, 0) per spatial dimension.

Skip any negative padding (i.e., no cropping).

Ensure bounding boxes and other geometric data keys remain unchanged when no padding is applied.


Add unit tests covering:
Image smaller than target → padding applied
Image equal to target → no-op
Image larger than target → no-op (no cropping)

Please let me know:

Whether a behavior change is acceptable, or
If you prefer a rename / documentation-only fix instead.
Once clarified, I will submit a PR accordingly.

**Comment by bw4sz** [comment]
I am not a maintainer, so we should wait for their interests.

**Comment by sidd-27** [comment]
hmm i agree the names are indeed a little misleading, @edgarriba  what should the correct approach be here?

**Comment by shijianjian** [comment]
IIRC, it is a bit intended, but I would admit it may be a big confusing. I think we may add a simple argument for this `PadTo(..., crop_if_exeeds=True)`.

**Comment by pragnyanramtha** [comment]
Would a `crop_if_exceeds: bool = True` parameter, keeping current behavior by default and adding tests/docs for `False`, match the intended fix?

**Comment by Buland-Choudhary** [comment]
Hi! I'd like to work on this if it's still open.

What's happening: PadTo.apply_transform ([kornia/augmentation/_2d/geometric/pad.py](https://github.com/nandikadivi-max/murder-mystery-ai/blob/main/kornia/augmentation/_2d/geometric/pad.py)) always calls

height_pad = flags["size"][0] - height
width_pad = flags["size"][1] - width
torch.nn.functional.pad(input, [0, width_pad, 0, height_pad], mode=..., value=...)
When the input is larger than size, height_pad/width_pad go negative. torch.nn.functional.pad treats a negative pad as a trim on that side, that's documented PyTorch behavior, not a bug in pad itself — so PadTo ends up silently cropping instead of only padding.

Plan, building on @shijianjian's suggestion above: add an opt-in crop_if_exceeds: bool = True flag to PadTo.__init__, defaulting to True so existing behavior (and anyone relying on it) doesn't change. When False, clamp height_pad/width_pad to max(..., 0) before calling F.pad, so an oversized input is left unchanged on that axis instead of being cropped. inverse_transform shouldn't need changes, slicing to the stored size is already a no-op when nothing was cropped.

I'll add tests for: input smaller than target (unchanged), input larger with crop_if_exceeds=True (regression test for current behavior), and larger with crop_if_exceeds=False (new — dimension preserved). Happy to update the docstring to spell out the crop behavior either way.

Could this be assigned to me?

**Comment by ducha-aiki** [comment]
Please, go ahead with crop_if_exceeds: bool = True

**Comment by victorwon2001** [comment]
The issue provides a concrete Kornia 0.8.2 reproduction showing that PadTo crops inputs larger than the requested target. Discussion indicates that this crop-to-fixed-size behavior is at least partly intentional, so the agreed direction is to preserve it by default while adding an opt-out argument, `crop_if_exceeds: bool = True`, with tests and documentation covering the non-cropping mode.

`subject`

> Issue: PadTo crops images when they are larger than the target size,
> instead of only padding when smaller.

`subject`

> Crops to target size, the image size is 1000, since this is target size.

`comment:4100080854`

> IIRC, it is a bit intended, but I would admit it may be a big confusing. I think we may add a simple argument for this `PadTo(..., crop_if_exeeds=True)`.

`comment:5231447616`

> Please, go ahead with crop_if_exceeds: bool = True

The supplied minimal reproduction was not executed, so its reported 1100x1100 to 1000x1000 result is extracted from the issue rather than independently verified.

No repository source files were supplied, so the current implementation details and exact test/doc locations were not independently assessed.

Proceed with the maintainer-approved `crop_if_exceeds: bool = True` direction: preserve the current crop behavior by default, allow oversized dimensions to remain unchanged when the option is false, and add tests/docs for both modes. The supplied reproduction has not been executed in this analysis.

SHA: `90650596d4de49448ea1359975d3baff71931c84`
